sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala

```diff
@@ -52,6 +52,23 @@ class DataTypeSuite extends SparkFunSuite {
     assert(StructField("b", LongType, false) === struct("b"))
   }
 
+  test("construct with add from StructField with comments") {
```

Comment:
Although more test is better, I think for this case we only need to test `withComment` and `getComment`, it's obvious that other `StructField` creation is calling `withComment`.